servo/fast_fail.py

```diff
@@ -104,6 +117,32 @@ def check_readings(
                 threshold_scalar = _get_scalar_from_readings(threshold_readings)
                 threshold_value = threshold_scalar * condition.threshold_multiplier
 
+                if self.config.zero_handling and float(threshold_value) == 0:
+                    self._results[condition].append(
+                        SloOutcome(
+                            **result_args, status=SloOutcomeStatus.missing_threshold
+                        )
+                    )
+                    continue
+
+                elif (float(threshold_value) == 0) and (
+                    0 < metric_value <= condition.slo_metric_minimum
+                ):
```

Comment:
Could you elaborate on the logic behind these checks? Seems to me like you only need the second part of the `and` and should be checking threshold_value instead of metric_value but maybe I'm missing something?